### PR TITLE
Allow overwrite docstring using raw json dict

### PR DESCRIPTION
**Pull request by BrikerMan**
Issue: https://github.com/python-restx/flask-restx/issues/48
Example

```python
POST_BODY_DOC = """
raw_doc_keys: 
  - consumes
  - parameters
consumes:
  - application/json
parameters:
  - in: body
    name: user
    description: The user to create.
    schema:
      type: object
      required:
        - userName
      properties:
        userName:
          type: string
        firstName:
          type: string
"""

@api .route('/app/demo')
class DemoAPI(Resource):
    @api.doc(**yaml.load(POST_BODY_DOC))
    def post(self):
          .....
```

### CODE REVIEW

**Comment by j5awry** [comment]
This is missing tests. Please add a test case passing in a dict and ensuring the swagger doc is set properly.

**Comment by BrikerMan** [comment]
@j5awry Test added~

**Comment by SVilgelm** [comment]
The CI fails due to:
```
>>> Python Static Analysis
1340
tests/test_swagger.py:1760:16: E127 continuation line over-indented for visual indent
1341
tests/test_swagger.py:1762:16: E127 continuation line over-indented for visual indent
1342
```
Please fix it

**Comment by SteadBytes** [comment]
Thank you for the contribution, however I am not in favour of this implementation. There is ongoing discussion over on https://github.com/python-restx/flask-restx/issues/48 around supporting an API of this *nature*. My main issues with this implementation:
- Likely to produce invalid Swagger specifications (no validation)
- Poor interop with the core `flask-restx` API for generating specs (`api.expect`, `api.marshal_with`) as they may simply be overwritten
- Further complicates the API for users e.g. making a decision between `api.expect` style decorators, passing a dict, does the dict version have the same effect as the decorators?, will the dict version overwrite anything else? How do I ensure the two are kept up to date? etc
  - Admittedly, the current API needs work and can be confusing (we're working on that so feedback is welcome!)

File: tests/test_swagger.py [hunk 1]
```diff
@@ -46,15 +46,15 @@ def output_xml(data, code, headers=None):
...
-            title='My API',
-            description='This is a testing API',
-            terms_url='http://somewhere.com/terms/',
-            contact='Support',
-            contact_url='http://support.somewhere.com',
-            contact_email='contact@somewhere.com',
-            license='Apache 2.0',
-            license_url='http://www.apache.org/licenses/LICENSE-2.0.html'
-        )
+                        title='My API',
```
Comment (SteadBytes) [inline tests/test_swagger.py]: Please avoid unrelated style changes :sweat_smile:

File: tests/test_swagger.py [hunk 2]
```diff
@@ -81,15 +81,15 @@ def test_specs_endpoint_info(self, app, client):
...
-            title='My API',
-            description='This is a testing API',
-            terms_url='http://somewhere.com/terms/',
-            contact='Support',
-            contact_url='http://support.somewhere.com',
-            contact_email='contact@somewhere.com',
-            license='Apache 2.0',
-            license_url='http://www.apache.org/licenses/LICENSE-2.0.html'
-        )
+                     title='My API',
```
Comment (SteadBytes) [inline tests/test_swagger.py]: https://github.com/python-restx/flask-restx/pull/49/files#r379792717

File: tests/test_swagger.py [hunk 3]
```diff
@@ -115,15 +115,15 @@ def test_specs_endpoint_info_delayed(self, app, client):
...
-            title=lambda: 'My API',
-            description=lambda: 'This is a testing API',
-            terms_url=lambda: 'http://somewhere.com/terms/',
-            contact=lambda: 'Support',
-            contact_url=lambda: 'http://support.somewhere.com',
-            contact_email=lambda: 'contact@somewhere.com',
-            license=lambda: 'Apache 2.0',
-            license_url=lambda: 'http://www.apache.org/licenses/LICENSE-2.0.html'
-        )
+                        title=lambda: 'My API',
```
Comment (SteadBytes) [inline tests/test_swagger.py]: https://github.com/python-restx/flask-restx/pull/49/files#r379792717

File: tests/test_swagger.py [hunk 4]
```diff
@@ -1425,17 +1490,17 @@ def get(self):
         assert description('get') == dedent('''\
             Parent description.
             Some details'''
-        )
+                                            )
```
Comment (SteadBytes) [inline tests/test_swagger.py]: https://github.com/python-restx/flask-restx/pull/49/files#r379792717

File: tests/test_swagger.py [hunk 5]
```diff
@@ -1692,9 +1757,9 @@ def post(self):
 
         path = data['paths']['/model-bad-uri/']
         assert path['get']['responses']['200']['schema']['$ref'] == \
-            '#/definitions/Person%2F%2F%3Flots%7B%7D%20of%20%26illegals%40%60'
+               '#/definitions/Person%2F%2F%3Flots%7B%7D%20of%20%26illegals%40%60'
```
Comment (SteadBytes) [inline tests/test_swagger.py]: https://github.com/python-restx/flask-restx/pull/49/files#r379792717

**Comment by BrikerMan** [comment]
@SteadBytes Fixed.

**Comment by BrikerMan** [comment]
@SteadBytes any update?

**Comment by SteadBytes** [comment]
@BrikerMan Apologies for lack of activity, I think we've all had quite busy lives lately! :sweat_smile: 

As I stated previously (here and on #48 ) *I* do not think this fits in with the `flask-restx` approach and existing API. If another maintainer agrees/disagrees we can either close or continue this PR and #48. @j5awry what do you think?

Either way, I'd just like to point out that we really appreciate your contribution and if this doesn't get merged it is not a reflection on your efforts - just that the feature wasn't quite suited for `flask-restx` :smile:

**Comment by BrikerMan** [comment]
Hi, @SteadBytes thanks for the great repo, it helped us a lot. But for new projects, we had moved to fastapi. Thanks a lot for the great works. Closed this PR.
